### PR TITLE
Send a "connected" event when done connecting to kinesis

### PR DESCRIPTION
**Pull request by maxedmands**
Hi 👋 ! Thanks for this library, it's a really low-friction way to experiment with Kinesis / verify that things are working :)

Small change request:

We would like to know when the consumer has successfully completed its handshake with AWS and confirmed that the stream is capable of receiving messages. (This is useful, for instance, in a before hook for an inter-process smoke test that talks to AWS directly.)

(FYI @wwwidmer)

### CODE REVIEW

File: test/index.spec.js [hunk 1]
```diff
@@ -148,6 +148,19 @@ describe('main', () => {
...
+      it('emits connected event when successfully connected', () => {
+        client.describeStream = AWSPromise.resolve({StreamDescription: {Shards: [{ShardId: 'shard id'}]}})
+        client.getShardIterator = AWSPromise.resolve({ShardIterator: 'shard iterator'})
+
+        sinon.stub(main.KinesisStreamReader.prototype, 'readShard')
+        const reader = new main.KinesisStreamReader(client, 'stream name', {foo: 'bar'})
+
+        return new Promise((resolve) => {
+          reader.once('connected', resolve)
+          reader._startKinesis()
```
Comment (crccheck) [inline test/index.spec.js]: I can see why it's structured this way, and testing events in Mocha is really messy, but this test structure is hard to follow. I may revisit this in the future.

File: cli.js [hunk 2]
```diff
@@ -54,6 +55,7 @@ program
       options.regexFilter = '.*'
     }
     const reader = new index.KinesisStreamReader(client, streamName, options)
+    if (program.verbose) reader.on("connected", () => process.stdout.write(`Connected to stream: ${streamName}\n`))
```
[the remark this answers is not in the document]
Comment (demands) [reply cli.js]: Apologies for the delay in response! I suppose `process.stdout.write()` just seemed like a more explicit statement to me that we are adding output to the same stream that's also logging the kinesis messages.

[the remark this answers is not in the document]
Comment (demands) [reply cli.js]: As far as writing to stderr, that seems acceptable. Is that standard practice?

File: test/index.spec.js (hunk 1 above)
Comment (demands) [reply test/index.spec.js]: For sure! There's some indirection here that I wasn't sure how to avoid. I don't feel attached to this code in any way. :)

**Comment by crccheck** [comment]
meee too. I'm pretty busy for a few more days

File: cli.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (crccheck) [reply cli.js]: How I think about it is if you did `kinesis-console-consumer | jq` or `kinesis-console-consumer > file.txt`, would you want the `connected` message to show? If not, then stderr is better
